commit 61170596e1c0715519720947a9a3b8249a729101
Author: Eric Farman <farman@linux.ibm.com>
Date:   Fri Nov 9 03:39:29 2018 +0100

    s390/cio: Fix cleanup when unsupported IDA format is used
    
    [ Upstream commit b89e242eee8d4cd8261d8d821c62c5d1efc454d0 ]
    
    Direct returns from within a loop are rude, but it doesn't mean it gets
    to avoid releasing the memory acquired beforehand.
    
    Signed-off-by: Eric Farman <farman@linux.ibm.com>
    Message-Id: <20181109023937.96105-3-farman@linux.ibm.com>
    Reviewed-by: Farhan Ali <alifm@linux.ibm.com>
    Reviewed-by: Pierre Morel <pmorel@linux.ibm.com>
    Acked-by: Halil Pasic <pasic@linux.ibm.com>
    Signed-off-by: Cornelia Huck <cohuck@redhat.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/s390/cio/vfio_ccw_cp.c b/drivers/s390/cio/vfio_ccw_cp.c
index ef5ab45d94b3..70a006ba4d05 100644
--- a/drivers/s390/cio/vfio_ccw_cp.c
+++ b/drivers/s390/cio/vfio_ccw_cp.c
@@ -387,8 +387,10 @@ static int ccwchain_calc_length(u64 iova, struct channel_program *cp)
 		 * orb specified one of the unsupported formats, we defer
 		 * checking for IDAWs in unsupported formats to here.
 		 */
-		if ((!cp->orb.cmd.c64 || cp->orb.cmd.i2k) && ccw_is_idal(ccw))
+		if ((!cp->orb.cmd.c64 || cp->orb.cmd.i2k) && ccw_is_idal(ccw)) {
+			kfree(p);
 			return -EOPNOTSUPP;
+		}
 
 		if ((!ccw_is_chain(ccw)) && (!ccw_is_tic(ccw)))
 			break;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1044_61170596e1c0715519720947a9a3b8249a729101.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1044_61170596e1c0715519720947a9a3b8249a729101.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
